src/test/java/org/hyperledger/fabric/sdk/NetworkConfigTest.java

```diff
@@ -318,6 +320,61 @@ public void testLoadFromConfigFileYamlNOOverrides() throws Exception {
 
     }
 
+    @Test
+    public void testLoadFromConfigFileJsonNOOverridesClientTls() throws Exception {
+
+        File f = NETWORK_CONFIG_CLIENT_TLS_JSON.toFile();
+        NetworkConfig config = NetworkConfig.fromJsonFile(f);
+
+        assertNotNull(config);
+
+        String peer0 = "peer0.org1.example.com";
+        String peer1 = "peer1.org1.example.com";
+        // Check JsonPeers' properties
+        Properties peer0Properties = config.getPeerProperties(peer0);
+        assertEquals(peer0Properties.getProperty("clientCertFile"), "./tls/sign.pem");
+        assertEquals(peer0Properties.getProperty("clientKeyFile"), "./tls/key.pem");
+        assertNull(peer0Properties.getProperty("tlsClientCertFile"));
+        assertNull(peer0Properties.getProperty("tlsClientKeyFile"));
+
+        Properties peer1Properties = config.getPeerProperties(peer1);
+        byte[] clientKeyBytes = (byte[]) peer1Properties.get("clientKeyBytes");
+        byte[] clientCertBytes = (byte[]) peer1Properties.get("clientCertBytes");
+        assertTrue(Arrays.equals(clientCertBytes, "-----BEGIN CERTIFICATE----- <etc>".getBytes()));
+        assertTrue(Arrays.equals(clientKeyBytes, "-----BEGIN PRIVATE KEY----- <etc>".getBytes()));
+        assertNull(peer1Properties.getProperty("tlsClientCertBytes"));
+        assertNull(peer1Properties.getProperty("tlsClientKeyBytes"));
+
+        // Check Peer nodes' properties
+        HFClient client = HFClient.createNewInstance();
+        client.setCryptoSuite(CryptoSuite.Factory.getCryptoSuite());
+        client.setUserContext(TestUtils.getMockUser(USER_NAME, USER_MSP_ID));
+
+        Channel channel = client.loadChannelFromConfig("mychannel", config);
+        assertNotNull(channel);
+
+        assertFalse(channel.getPeers().isEmpty());
+
+        for (Peer peer : channel.getPeers()) {
+            Properties properties = peer.getProperties();
+            assertNotNull(properties);
+            if (peer.getName().equals(peer0)) {
+                assertEquals(properties.getProperty("clientCertFile"), "./tls/sign.pem");
+                assertEquals(properties.getProperty("clientKeyFile"), "./tls/key.pem");
+                assertNull(properties.getProperty("tlsClientCertFile"));
+                assertNull(properties.getProperty("tlsClientKeyFile"));
+            } else if (peer.getName().equals(peer1)) {
+                byte[] nodeClientKeyBytes = (byte[]) properties.get("clientKeyBytes");
+                byte[] nodeClientCertBytes = (byte[]) properties.get("clientCertBytes");
```

Comment:
I think it would be better to use constants from the code for these values.

Reply:
Replaced after moving constants to NetworkConfig.